backend/src/server/routes/v1/certificate-router.ts

```diff
@@ -92,45 +108,48 @@ export const registerCertificateRouter = async (server: FastifyZodProvider) => {
       body: z
         .object({
           profileId: z.string().uuid(),
-          projectId: z.string().uuid(),
-          commonName: validateTemplateRegexField.optional(),
-          keyUsages: z.nativeEnum(CertKeyUsageType).array().optional(),
-          extendedKeyUsages: z.nativeEnum(CertExtendedKeyUsageType).array().optional(),
-          altNames: z
-            .array(
-              z.object({
-                type: z.nativeEnum(CertSubjectAlternativeNameType),
-                value: z.string().min(1, "SAN value cannot be empty")
-              })
-            )
-            .optional(),
-          signatureAlgorithm: z.nativeEnum(CertSignatureAlgorithm),
-          keyAlgorithm: z.nativeEnum(CertKeyAlgorithm),
           csr: z
             .string()
             .trim()
             .min(1, "CSR cannot be empty")
             .max(4096, "CSR cannot exceed 4096 characters")
             .optional(),
-          subjectAlternativeNames: z
-            .array(
-              z.object({
-                type: z.nativeEnum(ACMESANType),
-                value: z
-                  .string()
-                  .trim()
-                  .min(1, "SAN value cannot be empty")
-                  .max(255, "SAN value must be less than 255 characters")
-              })
-            )
+          attributes: z
+            .object({
+              commonName: validateTemplateRegexField.optional(),
+              keyUsages: z.nativeEnum(CertKeyUsageType).array().optional(),
+              extendedKeyUsages: z.nativeEnum(CertExtendedKeyUsageType).array().optional(),
+              altNames: z
+                .array(
+                  z.object({
+                    type: z.nativeEnum(CertSubjectAlternativeNameType),
+                    value: z.string().min(1, "SAN value cannot be empty")
+                  })
+                )
+                .optional(),
+              signatureAlgorithm: z.nativeEnum(CertSignatureAlgorithm).optional(),
+              keyAlgorithm: z.nativeEnum(CertKeyAlgorithm).optional(),
+              subjectAlternativeNames: z
+                .array(
+                  z.object({
+                    type: z.nativeEnum(ACMESANType),
+                    value: z
+                      .string()
+                      .trim()
+                      .min(1, "SAN value cannot be empty")
+                      .max(255, "SAN value must be less than 255 characters")
+                  })
+                )
+                .optional(),
+              ttl: z
+                .string()
+                .trim()
+                .min(1, "TTL cannot be empty")
+                .refine((val) => ms(val) > 0, "TTL must be a positive number"),
+              notBefore: validateCaDateField.optional(),
+              notAfter: validateCaDateField.optional()
```

Comment:
i'm not sure if this should be in the attributes object because I think we'd still have to pass these in even if use-case is just CSR?